app/models/services/crossref.rb

```diff
@@ -0,0 +1,17 @@
+class Crossref
+  def initialize(options={})
+    @base_url = 'http://api.crossref.org/'
+    @metadata_url = @base_url + "works/"
+    @funder_url = @base_url + "funders/"
+  end
+
+  def get_doi_metadata(doi)
+    metadata_uri = URI.parse(@metadata_url+doi)
+    begin
+      JSON.parse Net::HTTP.get(metadata_uri)
+    rescue JSON::ParserError => e
+      return nil
+    end
```

Comment:
This isn't really an object - in this case, it's a return value from an HTTP request. On our side, it's just data, or a series of attributes. We're not retrieving a record or interacting with some object. In this case, to handle the error of a bad DOI like http://api.crossref.org/works/10.1037/poop, we can just check the HTTP status code. If you open up the network tab while loading that URL, you can see that it's a 404. We can check to see if an HTTP request returns a success code (2xx range) or perhaps a redirect and then a 200 if we're getting fancy, and then handle the result accordingly.

Also, we should use rescue blocks sparingly. It's not really an application level breakage if an HTTP request returns with a non-success status code. https://webcache.googleusercontent.com/search?q=cache:9LlMiCOwyOkJ:https://httpstatusdogs.com/+&cd=1&hl=en&ct=clnk&gl=us